src/coreComponents/physicsSolvers/fluidFlow/ThermalCompositionalMultiphaseFVMKernels.hpp

```diff
@@ -0,0 +1,685 @@
+
+      // Step 3.2: compute the derivative of component flux wrt temperature
+
+      // slice some constitutive arrays to avoid too much indexing in component loop
+      arraySlice1d< real64 const, multifluid::USD_PHASE_COMP - 3 > phaseCompFracSub =
+        m_phaseCompFrac[er_up][esr_up][ei_up][0][ip];
+      arraySlice2d< real64 const, multifluid::USD_PHASE_COMP_DC - 3 > dPhaseCompFracSub =
+        m_dPhaseCompFrac[er_up][esr_up][ei_up][0][ip];
+
+      for( integer ic = 0; ic < numComp; ++ic )
+      {
+        real64 const ycp = phaseCompFracSub[ic];
+        for( integer ke = 0; ke < stack.stencilSize; ++ke )
+        {
+          dCompFlux_dTSlice[ke][ic] += dPhaseFlux_dT[ke] * ycp;
+        }
+        dCompFlux_dTSlice[k_up][ic] += phaseFlux * dPhaseCompFracSub[ic][Deriv::dT];
+      }
+
+      // Step 4: compute the enthalpy flux
+
+      real64 const enthalpy = m_phaseEnthalpy[er_up][esr_up][ei_up][0][ip];
+      *energyFluxPtr += phaseFlux * enthalpy;
+
+      for( integer ke = 0; ke < stack.stencilSize; ++ke )
+      {
+        dEnergyFlux_dPSlice[ke] += dPhaseFlux_dP[ke] * enthalpy;
+        dEnergyFlux_dTSlice[ke] += dPhaseFlux_dT[ke] * enthalpy;
+
+        for( integer jc = 0; jc < numComp; ++jc )
+        {
+          dEnergyFlux_dCSlice[ke][jc] += dPhaseFlux_dC[ke][jc] * enthalpy;
+        }
+      }
+
+      dEnergyFlux_dPSlice[k_up] += phaseFlux * m_dPhaseEnthalpy[er_up][esr_up][ei_up][0][ip][Deriv::dP];
+      dEnergyFlux_dTSlice[k_up] += phaseFlux * m_dPhaseEnthalpy[er_up][esr_up][ei_up][0][ip][Deriv::dT];
+
+      real64 dProp_dC[numComp]{};
+      applyChainRule( numComp,
+                      m_dCompFrac_dCompDens[er_up][esr_up][ei_up],
+                      m_dPhaseEnthalpy[er_up][esr_up][ei_up][0][ip],
+                      dProp_dC,
+                      Deriv::dC );
+      for( integer jc = 0; jc < numComp; ++jc )
+      {
+        dEnergyFlux_dCSlice[k_up][jc] += phaseFlux * dProp_dC[jc];
+      }
+    } );
+
+    // *****************************************************
+    // Computation of the conduction term in the energy flux
+    // Note that this term is computed using an explicit treatment of conductivity for now
+    // Step 1: compute the thermal transmissibilities at this face
+    m_stencilWrapper.computeWeights( iconn,
+                                     m_thermalConductivity,
+                                     m_thermalConductivity, // unused for now
+                                     stack.thermalTransmissibility,
+                                     stack.dTrans_dPres ); // unused for now
+
+    // Step 2: compute temperature difference at the interface
+    for( integer i = 0; i < stack.stencilSize; ++i )
+    {
+      localIndex const er  = m_seri( iconn, i );
+      localIndex const esr = m_sesri( iconn, i );
+      localIndex const ei  = m_sei( iconn, i );
+
+      stack.energyFlux += stack.thermalTransmissibility[0][i] * ( m_temp[er][esr][ei] + m_dTemp[er][esr][ei] );
```

Comment:
this is only the conductive part right?

Reply:
Yes you are right, this is only the conductive part here, added to the energy flux. The enthalpy flux transported by the phases is computed and added to the energy flux in the lambda above. I added some comments to clarify that.